src/driver/resnet50.cpp

```diff
@@ -25,1453 +25,912 @@
-    auto x_0                   = mmain->add_parameter(
-        "0", migraphx::shape{migraphx::shape::float_type, {batch, 3, 224, 224}});
-    auto x_main_module_2 = mmain->add_literal(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {1000}}, 1));
+    auto x_main_module_1         = mmain->add_literal(migraphx::abs(
+        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {1}}, 1)));
+    auto x_input_tensor_module_0 = mmain->add_parameter(
+        "input_tensor:0", migraphx::shape{migraphx::shape::float_type, {batch, 3, 224, 224}});
     auto x_main_module_3 = mmain->add_literal(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {1000, 2048}}, 2));
-    auto x_main_module_4 = mmain->add_literal(migraphx::abs(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {2048}}, 3)));
+        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 2));
+    auto x_main_module_4 = mmain->add_literal(migraphx::generate_literal(
+        migraphx::shape{migraphx::shape::float_type, {512, 512, 3, 3}}, 3));
     auto x_main_module_5 = mmain->add_literal(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {2048}}, 4));
-    auto x_main_module_6 = mmain->add_literal(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {2048}}, 5));
-    auto x_main_module_7  = mmain->add_literal(migraphx::abs(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {2048}}, 6)));
-    auto x_main_module_8  = mmain->add_literal(migraphx::generate_literal(
-        migraphx::shape{migraphx::shape::float_type, {2048, 512, 1, 1}}, 7));
-    auto x_main_module_9  = mmain->add_literal(migraphx::abs(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 8)));
-    auto x_main_module_10 = mmain->add_literal(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 9));
-    auto x_main_module_11 = mmain->add_literal(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 10));
-    auto x_main_module_12 = mmain->add_literal(migraphx::abs(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 11)));
-    auto x_main_module_13 = mmain->add_literal(migraphx::generate_literal(
-        migraphx::shape{migraphx::shape::float_type, {512, 512, 3, 3}}, 12));
-    auto x_main_module_14 = mmain->add_literal(migraphx::abs(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 13)));
-    auto x_main_module_15 = mmain->add_literal(
-        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 14));
+        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {512}}, 4));
+    auto x_main_module_6 = mmain->add_literal(migraphx::generate_literal(
```

Comment:
Not for this PR, but i guess we can write these models similar to, 
https://github.com/keras-team/keras-applications/blob/master/keras_applications/resnet50.py

In a modularized way.